packages/rich-text/src/component/index.native.js

```diff
@@ -346,93 +265,62 @@ export class RichText extends Component {
 		this.lastAztecEventType = 'content size change';
 	}
 
-	onEnter( event ) {
-		if ( this.props.onEnter ) {
-			this.props.onEnter();
+	handleEnter( event ) {
+		const { onEnter } = this.props;
+
+		if ( ! onEnter ) {
 			return;
 		}
-		const {
-			__unstableOnReplace: onReplace,
-			__unstableOnSplit: onSplit,
-		} = this.props;
-
-		this.lastEventCount = event.nativeEvent.eventCount;
-		this.comesFromAztec = true;
-		this.firedAfterTextChanged = event.nativeEvent.firedAfterTextChanged;
 
-		const canSplit = onReplace && onSplit;
-		const currentRecord = this.createRecord();
-		if ( this.multilineTag ) {
-			if ( event.shiftKey ) {
-				this.needsSelectionUpdate = true;
-				const insertedLineBreak = { ...insert( currentRecord, '\n' ) };
-				this.onFormatChange( insertedLineBreak );
-			} else if ( canSplit && isEmptyLine( currentRecord ) ) {
-				this.onSplit( currentRecord );
-			} else {
-				this.needsSelectionUpdate = true;
-				const insertedLineSeparator = { ...insertLineSeparator( currentRecord ) };
-				this.onFormatChange( insertedLineSeparator );
-			}
-		} else if ( event.shiftKey || ! onSplit ) {
-			this.needsSelectionUpdate = true;
-			const insertedLineBreak = { ...insert( currentRecord, '\n' ) };
-			this.onFormatChange( insertedLineBreak );
-		} else {
-			this.onSplit( currentRecord );
-		}
+		onEnter( {
+			value: this.createRecord(),
+			onChange: this.onFormatChange,
+			shiftKey: event.shiftKey,
+		} );
 		this.lastAztecEventType = 'input';
 	}
 
-	onBackspace( event ) {
-		const {
-			__unstableOnMerge: onMerge,
-			__unstableOnRemove: onRemove,
-			onChange,
-		} = this.props;
-		if ( ! onMerge && ! onRemove ) {
-			return;
-		}
-
+	handleDelete( event ) {
 		const keyCode = BACKSPACE; // TODO : should we differentiate BACKSPACE and DELETE?
```

Comment:
I don't think Aztec is currently capturing `DELETE` but it would be a nice thing to have as well. I don't see a way to inject the `keycode` prop into the event from the native side, but we could have `AztecView` do that if we want to keep the support for `DELETE` here.

Reply:
I was thinking to change the code on our AztecView to send keycodes the same way as the web, but we will only send up DELETE, ENTER and BACKSPACE for now.
 But I was considering that for another PR more specific to AztecView.

Reply:
👍 I think it's fine to leave it for later